swan-cmd/src/main/java/de/fraunhofer/iem/swan/io/dataset/InvokedMethodsFilter.java

```diff
@@ -0,0 +1,57 @@
+package de.fraunhofer.iem.swan.io.dataset;
+
+import de.fraunhofer.iem.swan.data.Method;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+/**
+ * Filters invoked methods in the application that are already part of the dataset
+ *
+ */
+
+public class InvokedMethodsFilter {
+
+    Set<Method> testMethods;
+    static Set<Method> invokedMethodsInDataset;
+    Set<Method> filteredTestMethods;
+    Set<Method> datasetMethods;
+    private static final Logger logger = LoggerFactory.getLogger(InvokedMethodsFilter.class);
+
+    public InvokedMethodsFilter(Set<Method> testMethods, Set<Method> datasetMethods){
+        this.testMethods = testMethods;
+        this.datasetMethods = datasetMethods;
+        invokedMethodsInDataset = new HashSet<>();
+        this.filteredTestMethods = new HashSet<>();
+    }
+
+    public Set<Method> filterUnknownInvokedMethods(){
+        for(Method method : testMethods){
+            if(method.isApplicationMethod()){
+                filteredTestMethods.add(method);
+                continue;
+            }
+            boolean inDataset = false;
+            for(Method datasetMethod : datasetMethods){
+                String testMethodSignature = method.getTrimmedSignature();
+                if(datasetMethod.getTrimmedSignature().equals((testMethodSignature))){
```

Comment:
There is an alternative way to do this by implementing the .contains() method properly. Not something you need to change for now but something you can take a look at.